README.md

```diff
@@ -386,6 +453,36 @@ TODO - explain how subqueries work
 Check out the [zenodbdemo](zenodbdemo/zenodbdemo.go) for an example of how to
 embed zenodb.
 
+## Implementation Notes
+
+### Sequences
+
+These are the central units of data storage in Zenodb: https://github.com/getlantern/zenodb/blob/master/encoding/seq.go
+
+In short, we group by dimension, and then we summarize/roll-up fields. The summarization is done using an aggregation function like SUM, AVG, MIN, MAX, etc. That is key aspect to how Zenodb manages to discard data and achieve compression.
+For example, for a new view, let’s say we’re not grouping by anything (that would be `GROUP BY _, PERIOD(1h)`), and let’s look just at the success_count field, in storage, we would have a single row with no dimensions and an array of aggregated success_counts by time period:
+
+`success_count: [4000, 5000]`
+
+That would be the result of a series of points that add up to 4000 in the first period, and to 5000 in the second period (of 1h)
+
+Now let’s say that we did `GROUP BY geo_country, PERIOD(1h)`
+
+And let’s say that success_counts are evenly split between US and AU, then we would have two rows:
+
+```
+geo_country: US   success_count: [2000, 2500]
+geo_country: AU   success_count: [2000, 2500]
+```
+
+One important aspect is that each field is actually not just an array, the array is actually preceded by the timestamp high water mark, so it would actually be something like this:
+
+`geo_country: US   success_count: 20180118T05:00Z[1000, 1500]`
+
+### Views
+
+Since everything in zenodb comes in through input streams, views cannot actually be constructed from the underlying tables, so they need to be stored independently. This allows for views to have different granularities that the tables/views they are referring to. In other words, at runtime, views are actually just like tables that pull from that same input stream, the only difference is that when you define a view, it can take into account knowledge from the definition of the underlying table.
```

Comment:
Can views get data directly from the input stream, or only from other views/tables?

Reply:
What @uaalto is trying to explain here is that views actually can get data only from the input stream. It's impossible to retroactively populate a view using data from the underlying table. Views are really just a language construct for defining a table that's derived from another table.